### PR TITLE
[#148170489] Transactions filters

### PR DESCRIPTION
**Pull request by guillaumedebavelaere**
created on top of https://github.com/tradingticket/TradeItIosTicketSDK2/pull/311

### CODE REVIEW

File: TradeItIosTicketSDK2/TradeItTransactionsTableViewManager.swift [hunk 1]
```diff
@@ -125,9 +147,21 @@ fileprivate class TransactionHistoryResultPresenter {
...
+        self.filterType = filterType
+        self.transactionsFiltered = self.transactions.filter {
+            TradeItTransactionPresenter($0, currencyCode: accountBaseCurrency).belongsToFilter(filter: filterType)
+        }
+    }
+    func numberOfTransactions(forFilterType filterType: TransactionFilterType) -> Int {
+        return self.transactions.filter {
+            TradeItTransactionPresenter($0, currencyCode: accountBaseCurrency).belongsToFilter(filter: filterType)
+        }.count
+    }
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItTransactionsTableViewManager.swift]: `filterTransactions(filterType: filterType).count`

File: TradeItIosTicketSDK2/TradeItTransactionsViewController.swift [hunk 2]
```diff
@@ -6,36 +6,89 @@ class TradeItTransactionsViewController: TradeItViewController, TradeItTransacti
...
+                alertController.addAction(otherAction)
+            }
+        }
+
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+        alertController.addAction(cancelAction)
+
+        if UIDevice.current.userInterfaceIdiom == .pad,
+            let popoverPresentationController = alertController.popoverPresentationController {
+            popoverPresentationController.barButtonItem = sender
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItTransactionsViewController.swift]: 👍

File: TradeItIosTicketSDK2/TradeItTransactionsViewController.swift [hunk 3]
```diff
@@ -6,36 +6,89 @@ class TradeItTransactionsViewController: TradeItViewController, TradeItTransacti
...
+                            securityQuestion,
+                            onViewController: self,
+                            onAnswerSecurityQuestion: answerSecurityQuestion,
+                            onCancelSecurityQuestion: cancelSecurityQuestion
+                        )
+                },
+                    onFailure: reject
+                )
+            }
+        }
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItTransactionsViewController.swift]: Use our new helper

File: TradeItIosTicketSDK2/TradeItTransactionsTableViewManager.swift [hunk 4]
```diff
@@ -17,17 +17,30 @@ class TradeItTransactionsTableViewManager: NSObject, UITableViewDelegate, UITabl
...
+        if let transactions = transactionHistoryResult.transactionHistoryDetailsList, !transactions.isEmpty {
+            self.transactionsTable?.backgroundView =  nil
+        } else {
+            self.transactionsTable?.backgroundView = noResultsBackgroundView
+        }
+        self.transactionsTable?.reloadData()
+    }
+
+    func filterTransactionHistoryResult(filterType: TransactionFilterType) {
+        self.filterType = filterType
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItTransactionsTableViewManager.swift]: Where does this get used? The `filterType` is also stored on the presenter. Would be nice to have only one place to keep track of this.

File: TradeItIosTicketSDK2/TradeItTransactionsViewController.swift [hunk 5]
```diff
@@ -6,36 +6,89 @@ class TradeItTransactionsViewController: TradeItViewController, TradeItTransacti
...
+            }
+            if transactionsPresenter.numberOfTransactions(forFilterType: .TRANSFERS) > 0 {
+                let transfersAction = provideTransactionsUIAlertAction(filterType: .TRANSFERS)
+                alertController.addAction(transfersAction)
+            }
+            if transactionsPresenter.numberOfTransactions(forFilterType: .FEES) > 0 {
+                let feesAction = provideTransactionsUIAlertAction(filterType: .FEES)
+                alertController.addAction(feesAction)
+            }
+            if transactionsPresenter.numberOfTransactions(forFilterType: .OTHER) > 0 {
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItTransactionsViewController.swift]: We're hitting this call a lot which is iterating over the transactions every time. Is there a nicer way to manage this?

Also might be able to dry this up similar to how we did on the other PR

**Comment by guillaumedebavelaere** [comment]
@jsom updated.
>We're hitting this call a lot which is iterating over the transactions every time. Is there a nicer way to manage this?
It is pretty quick on the simulator. We could add some caching if necessary ?

**Pull request merged**